### DISCUSSION TITLE
xbps-src clang support?

### DISCUSSION THREAD

**Discussion by 15zemlekop** [void-linux/void-packages#45118] [Q&A] [answered]
What should I do if I want to use clang instead of gcc to build packages with xbps-src?

**Comment by luckpng** [comment]


**Reply by classabbyamp** [reply]
this is not the proper way to do it.

1. what is installed on the host system does not matter
2. what `/usr/bin/cc` is does not matter
3. `/etc/xbps/xbps-src.conf` is not a thing
4. that is not what `XBPS_CROSS_BASE` is for

Did you get this from chatgpt? If so, I recommend not using that to answer questions here in the future.

**Comment by classabbyamp** [comment, marked as answer]
In general, this is not really supported by xbps-src. You can build specific packages with clang by adding `clang` to the `hostmakedepends` of the package, putting `CC="clang"` in the template, and it will probably work. Cross-compiling with clang would probably take a lot more work.

**Reply by 15zemlekop** [reply]
Ok, thanks
